app/controllers/candidate_interface/sign_in_controller.rb

```diff
@@ -46,11 +47,15 @@ def authenticate
 
       if authentication_token&.still_valid?
         candidate = authentication_token.user
-        flash[:success] = t('apply_from_find.account_created_message') if candidate.last_signed_in_at.nil?
+        first_sign_in = candidate.last_signed_in_at.nil?
+        flash[:success] = t('apply_from_find.account_created_message') if first_sign_in
         sign_in(candidate, scope: :candidate)
         add_identity_to_log(candidate.id)
         authentication_token.use!
 
+        @local_user = get_local_user
+        send_candidate_sign_in_confirmation_email unless first_sign_in
```

Comment:
Would it be a bit clearer to pass `local_user` to this function explicitly rather than set it (relatively) globally for that function to pick up?

Reply:
Hmm just had a crack at this. The issue I'm gaving is that the `@local_user` variable is also being called within the `get_local_user` method itself. By replacing the global variable with an instance variable it breaks the method.

Wecould just leave things as is, unless you can think of a better way?

Reply:
Looking at this again, maybe we don't even need `local_user`, given that we already have a `current_candidate` facility and we only call the `send_candidate_sign_in_confirmation_email` once they're signed in. Maybe I'd add a `raise` in `send_candidate_sign_in_confirmation_email` tho?

```ruby
def send_candidate_sign_in_confirmation_email
  raise 'Tried to send a confirmation email to a nonexistent candidate' unless current_candidate

  return if cookies.signed[:sign_in_confirmation] == current_candidate.id
  # etc...
end
```